spinoso-string/src/ascii_string/impls.rs

```diff
@@ -0,0 +1,99 @@
+use alloc::borrow::Cow;
+use alloc::vec::Vec;
+use core::borrow::{Borrow, BorrowMut};
+use core::ops::{Deref, DerefMut};
+
+use super::AsciiString;
+
+impl From<Vec<u8>> for AsciiString {
+    #[inline]
+    fn from(content: Vec<u8>) -> Self {
+        Self::new(content)
+    }
+}
+
+impl<'a> From<&'a [u8]> for AsciiString {
+    #[inline]
+    fn from(content: &'a [u8]) -> Self {
+        Self::new(content.to_vec())
+    }
+}
+
+impl<'a> From<&'a mut [u8]> for AsciiString {
+    #[inline]
+    fn from(content: &'a mut [u8]) -> Self {
+        Self::new(content.to_vec())
+    }
+}
+
+impl<'a> From<Cow<'a, [u8]>> for AsciiString {
+    #[inline]
+    fn from(content: Cow<'a, [u8]>) -> Self {
+        Self::new(content.into_owned())
+    }
+}
+
+impl From<alloc::string::String> for AsciiString {
+    #[inline]
+    fn from(s: alloc::string::String) -> Self {
+        Self::new(s.into_bytes())
+    }
+}
+
+impl From<&str> for AsciiString {
+    #[inline]
+    fn from(s: &str) -> Self {
+        Self::new(s.as_bytes().to_vec())
+    }
+}
+
+impl From<AsciiString> for Vec<u8> {
+    #[inline]
+    fn from(s: AsciiString) -> Self {
+        s.to_vec()
+    }
+}
+
+impl Deref for AsciiString {
+    type Target = [u8];
+
+    #[inline]
+    fn deref(&self) -> &[u8] {
+        &*self.as_slice()
```

Comment:
```suggestion
        self.as_slice()
```

Reply:
How do you feel about `fn deref(&self) -> &Self::Target`? (matches the rust documentation on deref)